src/orchestrator.ts

```diff
@@ -550,28 +542,28 @@ export class Orchestrator {
         }
     }
 
-    private all(state: HistoryEvent[], tasks: Task[]): TaskSet {
-        const faulted = tasks.filter(TaskFilter.isFailedTask);
-        if (faulted.length > 0) {
-            return TaskFactory.FailedTaskSet(tasks, (faulted[0] as FailedSingleTask).exception);
-        }
-
-        const succesfulTasks = tasks.filter(TaskFilter.isSuccessfulSingleTask);
-        const isCompleted = succesfulTasks.length === tasks.length;
-        if (isCompleted) {
-            const results = succesfulTasks.reduce((acc, t) => {
-                return [...acc, t.result];
-            }, []);
+    private all(state: HistoryEvent[], tasks: TaskBase[]): TaskSet {
+        const completedTasks = tasks.filter(TaskFilter.isCompletedTask);
+        if (completedTasks.length === tasks.length) {
+            const sortedTasks = completedTasks.sort(TaskFilter.CompareFinishedTime);
+            const completionIndex = sortedTasks[tasks.length - 1].completionIndex;
 
-            return TaskFactory.SuccessfulTaskSet(tasks, results);
+            const failedTasks = completedTasks.filter(TaskFilter.isFailedTask);
+            if (failedTasks.length > 0) {
+                // TODO: aggregate all failures into one clean error.
```

Comment:
Will this `TODO` be done as part of this PR or in another PR?

Also, is this `TODO` talking about the next line where we only consider the first exception and not any subsequent exceptions?

Reply:
I can do it as a part of this PR, just so it isn't forgotten about.

And yes, it is referring to next line. In C#, `Task.WhenAll()` accumulates all exceptions into an `AggregateException`, so I will have to do something similar here. Not sure how that would really really surface App Insights in the JS case though.